pkg/talos/client.go

```diff
@@ -47,6 +47,19 @@ func newClient(ctx context.Context, config *cloudConfig) (*client, error) {
 	}, nil
 }
 
+func (c *client) refreshClient(ctx context.Context) error {
+	if _, err := c.talos.Version(ctx); err != nil {
```

Comment:
I think we need to `.Close()` the old client `c.talos` (here or below on line 57), as otherwise we leak connections and these will be running forever in the client even if the new client is in use